### PR TITLE
Upgrade grpc and protobuf libraries

### PR DESCRIPTION
**Pull request by cocreature**
as requested by @danilofaria. The one thing still missing from this is the gRPC C++ libraries. Those are only used in our Haskell libs internally so while we should upgrade that, I care a bit less about doing that now.

changelog_begin

- [Dependencies] Upgrade
  iou.grpc:grpc-(api|core|netty|protobuf|services|stub) to
  1.35.0.

- [Dependencies] Upgrade
  io.netty:netty-(codec-http2|handler|handler-proxy|resolver)
  to 4.1.58.Final.

- [Dependencies] Upgrade
  io.netty:netty-tcnative-boringssl-static to 2.0.36.Final.

- [Dependencies] Upgrade com.google.protobuf:protobuf-java to 3.14.0.

changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/bindings-java/quickstart/template-root/pom.xml [hunk 1]
```diff
@@ -40,6 +40,11 @@
             <artifactId>spark-core</artifactId>
             <version>2.7.2</version>
         </dependency>
+        <dependency>
```
Comment (cocreature) [inline docs/source/app-dev/bindings-java/quickstart/template-root/pom.xml:43]: It looks like this is no longer in the classpath by virtual of being pulled in as a transitive dependency. Seems sensible to add it as a direct dep anyway if we rely on it.

**Comment by stefanobaghino-da** [comment]
it's happening

File: bazel-java-deps.bzl [hunk 2]
```diff
@@ -86,20 +86,20 @@ def install_java_deps():
...
-            "io.netty:netty-resolver:4.1.48.Final",
-            "io.netty:netty-tcnative-boringssl-static:2.0.30.Final",
+            "io.netty:netty-codec-http2:4.1.58.Final",
+            "io.netty:netty-handler:4.1.58.Final",
+            "io.netty:netty-handler-proxy:4.1.58.Final",
+            "io.netty:netty-resolver:4.1.58.Final",
+            "io.netty:netty-tcnative-boringssl-static:2.0.36.Final",
             # protobuf
-            "com.google.protobuf:protobuf-java:3.11.0",
+            "com.google.protobuf:protobuf-java:3.14.0",
```
Comment (stefanobaghino-da) [inline bazel-java-deps.bzl:102]: Does it make sense to define a few constants to change these in a single place?

**Review by stefanobaghino-da — APPROVED** [review]
LGTM, thanks for doing the work required to get to this point. 🙇🏻

File: bazel-java-deps.bzl (hunk 2 above)
Comment (cocreature) [reply bazel-java-deps.bzl]: good idea 84a8a9c287048eeab450143e8d3d0d7170ab5a76

File: compatibility/bazel_tools/daml_sdk.bzl [hunk 3]
```diff
@@ -68,6 +68,10 @@ def _daml_sdk_impl(ctx):
     ctx.extract(
         "ledger-api-test-tool.jar",
         output = "extracted-test-tool",
+        # We cannot fully extract the JAR because there are files
+        # that clash on case insensitive file systems. Luckily, we only
+        # need the DAR so this is not an issue.
+        stripPrefix = "ledger/test-common",
```
Comment (aherrmann-da) [inline compatibility/bazel_tools/daml_sdk.bzl:74]: Maybe something worth fixing in ledger-api-test-tool?

**Review by aherrmann-da — APPROVED** [review]
LGTM

File: compatibility/bazel_tools/daml_sdk.bzl (hunk 3 above)
Comment (cocreature) [reply compatibility/bazel_tools/daml_sdk.bzl:74]: I took a brief look at it but I couldn’t find a nice way to configure this.The deploy jars built by Bazel just bundle everything and jarjar only allows zapping classes not files in META-INF afaict.

**Pull request merged**